stack/services/src/main/java/org/apache/usergrid/security/shiro/utils/SubjectUtils.java

```diff
@@ -243,14 +243,18 @@ public static ApplicationInfo getApplication( Identifier identifier ) {
         String applicationName = null;
         UUID applicationId = null;
         BiMap<UUID, String> applications = getApplications();
+        Set<String> values= applications.values();
+
         if ( applications == null ) {
             return null;
         }
         if ( identifier.isName() ) {
             applicationName = identifier.getName().toLowerCase();
-            applicationId = applications.inverse().get( applicationName );
-            if ( applicationId == null ) {
-                applicationId = applications.inverse().get( identifier.getName() );
+            for (String value: values ){
```

Comment:
Dumb question, but why not just get from the bi-map after normalizing the string.  Then we'll have an O(1) lookup vs O(n)?